pkgs/development/go-modules/generic/default.nix

```diff
@@ -105,16 +107,18 @@ let
     inherit (go) GOOS GOARCH;
 
     GO111MODULE = "on";
+    GOFLAGS = "-mod=vendor";
 
     configurePhase = args.configurePhase or ''
       runHook preConfigure
 
       export GOCACHE=$TMPDIR/go-cache
       export GOPATH="$TMPDIR/go"
       export GOSUMDB=off
-      export GOPROXY=file://${go-modules}
-
+      export GOPROXY=off
       cd "$modRoot"
+      rm -rf vendor
+      cp -r ${go-modules} vendor
```

Comment:
External users are going to be confused when their program stop compiling because they still have the goModSha256 in their /nix/store.

Here I would add a check that looks at the go-modules content and complain if it's of the old format with an explanation. The format changed, please update the sha256.

Reply:
This might be a good alternative to renaming the attribute. Renaming could catch the error earlier though.

Reply:
Just did the rename instead - makes life easier.